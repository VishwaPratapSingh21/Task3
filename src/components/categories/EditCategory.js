import React ,{Component}from 'react';

/**
 * Component for handling editing categories.
 * @author [Hadijah kyampeire](https://github.com/hadijahkyampeire/Yummy_Reactjs_frontend)
 */
class EditCategory extends Component {
    state = {
        name:''
    }
    /**
     * @param {object} event - The element which handles edit category
     */
    handleEditCategory =(event) =>{
        event.preventDefault();
        this.props.editCategory(this.props.id, this.state.name)
    }
    handleInput =(event) =>{
        const {name, value} = event.target;
        this.setState({[name]:value});
    }

    // updates the state before rendering the component
    componentWillMount(){
        this.setState({name:this.props.name})
    }
    render() {
        return (
            <div
                className="modal fade"
                id={`edit_category${this.props.id}`}
                role="dialog"
                aria-labelledby="exampleModalLabel"
                aria-hidden="true">
                <div className="modal-dialog" role="document">
                    <div className="modal-content">
                        <div className="modal-header">
                            <h5 className="modal-title">Edit category name</h5>
                            <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                                <span aria-hidden="true">&times;</span>
                            </button>
                        </div>
                        <form onSubmit={this.handleEditCategory}>
                        <div className="modal-body">
                        {this.props.error ? (
                    <div className="alert alert-danger">{this.props.error}
                     <button
                    type="button"
                    class="close"
                    data-dismiss="alert"
                    aria-label="Close"
                  >
                    <span aria-hidden="true">&times;</span>
                  </button></div>
                  ) : (
                    ""
                  )}
                            <div className="form-group">
                                <input type="text" name="name" onChange={this.handleInput} className='form-control' value={this.state.name}/>
                            </div>

                        </div>
                        <div className="modal-footer">
                            <button type="submit" className="btn btn-primary" >Update</button>
                            <button type="button" className="btn btn-secondary" data-dismiss="modal" id={`close${this.props.id}`}>Cancel</button>
                        </div>
                        </form>
                    </div>
                </div>
            </div>
        );
    }
}
export default EditCategory;