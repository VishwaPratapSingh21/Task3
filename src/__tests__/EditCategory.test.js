import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import EditCategory from '../components/categories/EditCategory';

describe('EditCategory component', () => {
  const wrapper = shallow(<EditCategory editCategory={jest.fn()}/>);
  const preventDefault = jest.fn();

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has divs in a component', () => {
    expect(wrapper.find('div')).toHaveLength(7);
  });

  it('has headers h5 in a component', () => {
    expect(wrapper.find('h5')).toHaveLength(1);
  });

  it('has inputs in a modal', () => {
    expect(wrapper.find('input')).toHaveLength(1);
    expect(wrapper.find('form').simulate('submit', { preventDefault }))
    expect(preventDefault).toBeCalled();
    expect(wrapper.find('[name="name"]').simulate('change', { target: { name: 'name', value: 'lunch' }}));
  });

  it('has a form in a modal', () => {
    expect(wrapper.find('form')).toHaveLength(1);
  });

  it('has span in a modal', () => {
    expect(wrapper.find('span')).toHaveLength(1);
  });

  it('has button in a modal', () => {
    expect(wrapper.find('button')).toHaveLength(3);
  });

  it('calls component did mount', () => {
    sinon.spy(EditCategory.prototype, 'componentWillMount');
    const fullWrapper = mount(<EditCategory />);
    expect(EditCategory.prototype.componentWillMount).toHaveProperty('callCount', 1);
    EditCategory.prototype.componentWillMount.restore();
  });
});
