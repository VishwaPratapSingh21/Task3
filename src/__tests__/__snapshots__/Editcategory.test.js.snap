// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditCategory component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  id="edit_categoryundefined"
  role="dialog"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Edit category name
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="modal-body"
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="name"
              onChange={[Function]}
              type="text"
            />
          </div>
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary"
            type="submit"
          >
            Update
          </button>
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            id="closeundefined"
            type="button"
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
